internal/util/projutil/project_util.go

```diff
@@ -97,3 +88,27 @@ func GetOperatorType() OperatorType {
 	}
 	return OperatorTypeGo
 }
+
+func SaveGopath() string {
```

Comment:
Rename this to `GetGopath` and add something like the following:
```Go
func GetGopath() string {
    gopath, ok := os.Getenv(GopathEnv)
    if !ok || len(gopath) == 0 {
        log.Fatal("get current pkg failed: GOPATH env not set")
    }
    return gopath
}
```
You currently call `SaveGopath()` before `CheckAndGetCurrPkg()` is called, which checks if `GOPATH` is set. With the above change, the check will be performed correctly.

Remove [these lines](https://github.com/operator-framework/operator-sdk/blob/ef664ed8e1dfe7e14d8c19dbf3d439891fd4008b/internal/util/projutil/project_util.go#L77-L80) now that you check `GOPATH` in `main.go`.

Reply:
updated